### PR TITLE
Implement namespace creation/deletion logic

### LINKED ISSUE #43: Create namespaces if they don't exist
Namespaces are no longer implicitly handled and must exist before deploying services/deployments/etc.

### PR DESCRIPTION
**Pull request by stroebs**
Implement namespace creation/deletion logic:
- Implicitly create the target namespace if it doesn't exist
- Delete target namespace if it's empty with the `--cleanup` flag
- Use `yaml.safe_load` to avoid CLI warning

Closes https://github.com/EDITD/kubetools/issues/43

### CODE REVIEW

File: kubetools/deploy/commands/cleanup.py [hunk 1]
```diff
@@ -12,11 +14,14 @@
...
 # Handles removal of orphaned replicasets and pods as well as any complete jobs,
 # working on the namespace level only (no apps).
+# If the cleanup removes all remaining objects, the namespace will be deleted too.
 
 def get_cleanup_objects(build):
     replica_sets = list_replica_sets(build.env, build.namespace)
+    replica_set_names = set(get_object_name(replica_set) for replica_set in replica_sets)
     replica_sets_to_delete = []
     replica_set_names_to_delete = set()
+    replica_sets_already_deleted = set()
```
Comment (gchazot) [inline kubetools/deploy/commands/cleanup.py]:
```suggestion
    replica_set_names_already_deleted = set()
```
for consistency

File: kubetools/deploy/commands/cleanup.py [hunk 2]
```diff
@@ -26,33 +31,62 @@ def get_cleanup_objects(build):
...
             replica_sets_to_delete.append(replica_set)
 
+        if replicaset.metadata.deletion_timestamp:
+            replica_sets_already_deleted.add(get_object_name(replica_set))
+
     pods = list_pods(build.env, build.namespace)
+    pod_names = set(get_object_name(pod) for pod in pods)
     pods_to_delete = []
+    pod_names_to_delete = set()
+    pods_already_deleted = set()
```
Comment (gchazot) [inline kubetools/deploy/commands/cleanup.py]:
```suggestion
    pod_names_already_deleted = set()
```
for consistency

**Review by gchazot — APPROVED** [review]
LGTM, just a cosmetic suggestion

File: kubetools/deploy/commands/cleanup.py (hunk 1 above)
Comment (stroebs) [reply kubetools/deploy/commands/cleanup.py]: https://github.com/EDITD/kubetools/pull/58/commits/dd26efae5067be7cbdf034d726cbef6edb579a6e

File: kubetools/deploy/commands/cleanup.py (hunk 2 above)
Comment (stroebs) [reply kubetools/deploy/commands/cleanup.py]: https://github.com/EDITD/kubetools/pull/58/commits/dd26efae5067be7cbdf034d726cbef6edb579a6e

**Review by Fizzadar — APPROVED** [review]
LGTM pending changelog alignment!

**Review by ryan109 — APPROVED** [review]
👍

**Comment by stroebs** [comment]
ty @Fizzadar @gchazot @ryan109

**Pull request merged**